### ISSUE TITLE
GLArab HD stopped working? Any fixes

### ISSUE THREAD

**Issue by MatrixTV** [Shani-08/ShaniXBMCWork#230]
Hi Shani-8
I would like first to thank you for all your great work.
And want to know if there is a quick fix for GLArab HD? as it all stopped working in ShahidMBC.Net XMBC Client.
What have been changed?
Please let us know if there is a new update to fix it coming soon?
Thanks

**Comment by Shani-08** [comment]
Looks like they have patched their servers, i dont have access to their box etc so can't debug myself so waiting for someone to get us the logs. so far no updates, however, you have following options.
1. Use GLArab low quality option. not all channels are available.
2. Use windows PC and install GLProxy. then in the settings enable local proxy and it will work with HD. you can also use the same PC to serve other Non windows Systems as well.
3. someone create a VPS (hosted windows server) and install GL proxy and it will server everyone? group? etc

**Comment by allouch89** [comment]
Are you sure about the proxy? Because I tried it yesterday and it didn't work for me

**Comment by moe1999** [comment]
Yea. What if we have an android box. How are we supposed to get he proxy on there. And are you sure if the proxy works.

**Comment by Shani-08** [comment]
I just tested, make sure you use this proxy. glarab_http_proxy_1.4.0.5571 (download from here https://www.glarab.com:447/zixi/download/proxy/) 
Let me know the output, off course first test on the windows machine and make sure its working before you use android. On Android, you would need to enter the ip of the windows server.

**Comment by moe1999** [comment]
How would you enter the ip on the android

**Comment by Shani-08** [comment]
In the settings of Shahid! Look there in the GL section.

**Comment by hmemar** [comment]
Shani i van get vps for everyone to use. How do i install the proxy. Any instructions 

Regards

Husham Memar

**Comment by Shani-08** [comment]
just run the setup, get the link from above post, make sure port 4500 is opened. Let me know if you need help, pass me the ip of the server once you are done.

**Comment by moe1999** [comment]
Is there a link to get the proxy for android. Just like in Windows. I just downloaded it on Windows and it worked for me. I just need it now for android

Thanks

**Comment by Shani-08** [comment]
@moe1999  please read it again. There is no proxy for android, this is why you need to enter the IP address of windows machine on Android in Shahid Settings so that that windows machine as proxy.

**Comment by moe1999** [comment]
So where it says Gl Proxy (use 127.0.0.1) for local machine. You want me to put my laptops IP address

**Comment by Shani-08** [comment]
Yes, 127.0.0.1 means this machine. which is good for windows as the proxy is on the same machine. but on the android you have to enter the ip address of your laptop, make sure the firewall is off on windows

**Comment by allouch89** [comment]
I was so close at buying a glbox today, just needed to press "enter" but then i started thinking
Im not even sure how i would pull out the logs etc. from it

**Comment by Shani-08** [comment]
lol, we can help you do that :)
or wait for @hmemar to arrange the VPS he promised lol!

**Comment by hmemar** [comment]
the vps I have is a linux one and its not allwing the exe as you know I asked the ISP to get me a price on the vps for windows

**Comment by hmemar** [comment]
Hi shani - ok just paid for windows vps and they will send me the information soon
do you want me to give you control after wards to it to do it

**Comment by Shani-08** [comment]
Thanks. Yes send me login details on taacc14 at gM ail .com
Make sure to request to open the port 4500

**Comment by allouch89** [comment]
If the proxy keeps stop working and needs to restart it after a while (aprox. 1 day), what is wrong then?

**Comment by Shani-08** [comment]
dont know as its a bit of a blackbox. May be schedule a job to restart the window service?

**Comment by allouch89** [comment]
I guess that could help, but also sometimes it stops working within a day, maybe its being stressed out or something. Anyway thanks for your suggestion.

**Comment by kkflarg** [comment]
Hey shani, huge fan of your plugin, thank you for your work. The glarab source also stopped working for me, I just wanted to know if an update will be out for the addon to fix the issue, thanks!

**Comment by Shani-08** [comment]
@allouch89 if you want to try the latest version (i have patched it to work with network PCs), see http://ge.tt/1fBaeJP2 and follow instructions. Let me know if it works.

**Comment by Shani-08** [comment]
@kkflarg Both to read this issue from start and you should be able to get it working.
There are no fix available so you need proxy to setup.

**Comment by allouch89** [comment]
Thanks @Shani-08 i will try it now.

**Comment by allouch89** [comment]
@Shani-08 my pc is not allowing me to download the http_proxy.exe file, could you upload all files in a .rar / .zip file?

**Comment by allouch89** [comment]
Nvm, i managed to download it.
It is working, now i just hope that it will keep on working, i have removed the task schedule to see how long it will work without a restart. Thank you

**Comment by allouch89** [comment]
Btw, you can change the channel.flv to something else, like channel.mp4

**Comment by kkflarg** [comment]
@Shani-08 ok thanks, ill get on it.

**Comment by testiddd** [comment]
if you log in to .36 server with normal request you get m3u8 playlist - with renamed for example #EXTINF to #EXTZIXIINF , they use altered playlist for proxy ?

**Comment by kkflarg** [comment]
@Shani-08 it's working! :) but does my pc have to be on to use the proxy on my android tv?

**Comment by moe1999** [comment]
@kkflarg how do you get the proxy on android. I tried doing it but couldn't figure it out.

**Comment by kkflarg** [comment]
@moe1999 long press/click on the addon, a little popup window will come up. press addon settings, press the tab that has all the login info for the sources and scroll down to the GL login info section, att his point scroll to the bottom of the GL section and you will see a "Local Proxy" option, press that and where it says local machine will be ungreyed, then select that and enter your pc's ip. press ok.

**Comment by kkflarg** [comment]
@moe1999 Make sure you: 1. installed the GLProxy from above, opened port 4500, and since you want to use your pc's ip on an android you will most likely have to turn your router's firewall off.

**Comment by moe1999** [comment]
@kkflarg so all I do is put my laptops ip on where it says GL proxy for local machine ?

**Comment by moe1999** [comment]
How do you open the port

**Comment by allouch89** [comment]
with windows firewall and port forwarding in your routers controlpanel

**Comment by kkflarg** [comment]
@moe1999  yeah what @allouch89 said, all routers are different though, you will have to google how to do it for your specific router

**Comment by allouch89** [comment]
you can also contact your ISP and ask them how you can access your router to change/add some settings like port forwarding

however i think you can use it locally without port forwarding, just add the http_proxy.exe file to the exception list in your windows firewall and add the local ip to the settings in shahid settings for gl proxy. it should be something like 192.168...

**Comment by moe1999** [comment]
Oh ok that's great. I will try it right now!

**Comment by allouch89** [comment]
@Shani-08 the proxy that you sent, isnt working anymore, i need to restart it :(

**Comment by kkflarg** [comment]
I'm trying to figure out a way to make it work without taking down my firewall

**Comment by moe1999** [comment]
Yea if there's a way. That's better because it's not safe to take down the firewall

**Comment by allouch89** [comment]
Then do as i said, use port forwarding instead

**Comment by kkflarg** [comment]
its working fine on my pc, but my android tv seems to be blocked

**Comment by allouch89** [comment]
It's port 4500 if that's not obvious

**Comment by moe1999** [comment]
Where do I put the port!!!

**Comment by kkflarg** [comment]
@allouch89 i did forward the port, that is why it is working on my pc, my android, however, isnt working because the firewall is up

**Comment by kkflarg** [comment]
@moe1999 under the "port forwarding" section of your router's firewall settings

**Comment by allouch89** [comment]
If you did forward the port, then use your ip address, not your local one

**Comment by allouch89** [comment]
You can find your ip address here for example

**Comment by kkflarg** [comment]
@allouch89 yeah, I did that too. all that happens is that on the android tv, it flashes through all the ips for GLArab and then it fails

**Comment by allouch89** [comment]
http://YOUR.IP.ADDRESS:4500/json_version.htm
if you successfully set up the proxy with firewall and port forwarding, this should show something like:
proxy_found({"version" : "1.4.0.5571 32-bit", "major" : 1, "minor" : 4, "minor_minor":0, "build": 5571});

**Comment by kkflarg** [comment]
@allouch89 connection refused

**Comment by allouch89** [comment]
Did you add the http_proxy.exe file to windows firewall ?

**Comment by kkflarg** [comment]
@allouch89 I use avast, but ill try. what is strange is that it seems that the port forwarding worked since my pc is getting the channels

**Comment by allouch89** [comment]
your pc doesnt need port forwarding

**Comment by kkflarg** [comment]
oh...

**Comment by allouch89** [comment]
you need to allow the http_proxy.exe in windows firewall and also make sure its not blocked by avast

**Comment by kkflarg** [comment]
ok, where would that file be located?

**Comment by allouch89** [comment]
C:\Program Files\GLArab.com\Proxy\..

**Comment by kkflarg** [comment]
ok

**Comment by kkflarg** [comment]
i did it, Failure still occurring in the addon

**Comment by allouch89** [comment]
check with this link instead
http://YOUR.IP.ADDRESS:4500/json_version.htm faster to debug evertime you make changes

**Comment by allouch89** [comment]
i could try to check your settings with teamviewer if you'd want me to

**Comment by kkflarg** [comment]
connection is still refused, and I'd rather not sorry :/

**Comment by kkflarg** [comment]
will I have to keep my pc on whenever I'd want to use the tv on proxy? if so i might as well stop now.

**Comment by allouch89** [comment]
Yes you do, of course

**Comment by kkflarg** [comment]
@allouch89 ok, haha, I figured so. I guess ill have to buy a pc for my tv then. thanks for your time!

**Comment by kkflarg** [comment]
the best option for me will probably be what @hmemar was doing for us.

**Comment by allouch89** [comment]
yes but i dont think it will run good, when there is many users using the same proxy

**Comment by kkflarg** [comment]
yeah, it isnt, it keeps lagging and stopping.

**Comment by Shani-08** [comment]
Guys first make sure you understand why and what you guys doing as above conversation will confuse everyone. 
First thing first. On windows since you are able to run the http proxy, it works fine in Kodi as you are using local machine. No port need to be opened or forwarded. 
Now, since on android you cannot run the proxy, you want to use windows machine as your proxy to execute GL Zixi code. Obviously, the Windows PC required to be ON when you play channels on Android. 
Now when Android try to play, it communicates with windows machine, both of which are in your same network that is your LAN, hence if you either turn off the firewall on windows machine OR open the port 4500 on windows machine it will work. Remember, it's nothing to do with router or anything. There is NO port forwarding required. 
port forwarding in only required when you are not at Home, where the windows PC is running and you want your Mobile to run Kodi. This is where you need port forwarding on router to send the traffic to the windows machine and in your mobile you would need your PubLic IP not internal IP but this Is no is doing as everyone is setting up just inside their houses.

**Comment by kkflarg** [comment]
@Shani-08  ok, thanks for clearing things up.

**Comment by Shani-08** [comment]
I believe they have closed the loophole of proxy too, now it works for 10 seconds only. 
So without the glbox, we are at the dead end.

**Comment by kkflarg** [comment]
:disappointed:

**Comment by Shani-08** [comment]
Proxy HD should be working if you click on download. Not sure for how long but give it a try. but prepare for it to be stopped :)

**Comment by MatrixTV** [comment]
Thanks @Shani-08 for the update it looks like its working temporary as you mentioned :)

**Comment by hmemar** [comment]
Windows vps details sent to you.
Lets get proxys done ;-)

Regards

Husham Memar

**Comment by Shani-08** [comment]
Thanks @hmemar The port is closed,
in the meantime, i have some update.

OK - Try version 3.6.4 version and Non Proxy links should work too but for how long? probably not much long Smile
If you get error of dependency then make sure you have the repo installed. let me know.

**Comment by MatrixTV** [comment]
Thank you @Shani-08 and @hmemar (Husham) for working to resolve this issue. Its all appreciated.

**Comment by kkflarg** [comment]
:smile:

**Comment by MatrixTV** [comment]
It looks like GLArab HD working again with non windows as well :) but freezing more often.. any changes.

**Comment by ehaab** [comment]
@Shani-08 how to play glarab hd m3u8 on iphone?

**Comment by moe1999** [comment]
Glarab hd m3u does not work. They changed the way of streams. Now they have a start time.
